### ISSUE TITLE
Autocomplete field options in the SingleChildScrollView remains open even when field goes out of view.

### ISSUE THREAD

**Issue by nick9822** [flutter/flutter#154515]
### Steps to reproduce

1. Wrap Autocomplete field in the SingleChildScrollView.
2. Click on the field as if you are selecting options.
3. Scroll to make field out of view. Remember to scroll by two fingers (not by clicking and dragging the scrollbar).
4. You will notice the options overlay remains open and follow the field which is not even visible.

### Expected results

Options should close as soon as field goes out of view.

### Actual results

It remains open.

### Code sample

https://dartpad.dev/?id=61735dfd0bbd1de1211510d4cdab387d

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

![image](https://github.com/user-attachments/assets/0e2cb92b-984f-4d2a-824f-2976852295af)

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Dart SDK 3.5.1 and Flutter SDK 3.24.1
```

</details>

**Comment by darshankawar** [comment]
Thanks for the report. I was able to replicate this on web and desktop platforms, as shown below:


https://github.com/user-attachments/assets/27f8aa73-0cd7-4704-a6e9-5e4b47150dbf

```
Stable : 3.24.1
Master: 3.25.0-1.0.pre.219
```

<details>
<summary> code </summary>

```

import 'dart:async';

import 'package:flutter/material.dart';

/// Flutter code sample for [Autocomplete] that demonstrates fetching the
/// options asynchronously and debouncing the network calls, including handling
/// network errors.

void main() => runApp(const AutocompleteExampleApp());

const Duration fakeAPIDuration = Duration(seconds: 1);
const Duration debounceDuration = Duration(milliseconds: 500);

class AutocompleteExampleApp extends StatelessWidget {
  const AutocompleteExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text(
              'Autocomplete - async, debouncing, and network errors'),
        ),
        body: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              Text(
                  'Type below to autocomplete the following possible results: ${_FakeAPI._kOptions}.'),
              const SizedBox(height: 32.0),
              const _AsyncAutocomplete(),
            ],
          ),
        ),
      ),
    );
  }
}

class _AsyncAutocomplete extends StatefulWidget {
  const _AsyncAutocomplete();

  @override
  State<_AsyncAutocomplete> createState() => _AsyncAutocompleteState();
}

class _AsyncAutocompleteState extends State<_AsyncAutocomplete> {
  // The query currently being searched for. If null, there is no pending
  // request.
  String? _currentQuery;

  // The most recent options received from the API.
  late Iterable<String> _lastOptions = <String>[];

  late final _Debounceable<Iterable<String>?, String> _debouncedSearch;

  // Whether to consider the fake network to be offline.
  bool _networkEnabled = true;

  // A network error was received on the most recent query.
  bool _networkError = false;

  // Calls the "remote" API to search with the given query. Returns null when
  // the call has been made obsolete.
  Future<Iterable<String>?> _search(String query) async {
    _currentQuery = query;

    late final Iterable<String> options;
    try {
      options = await _FakeAPI.search(_currentQuery!, _networkEnabled);
    } catch (error) {
      if (error is _NetworkException) {
        setState(() {
          _networkError = true;
        });
        return <String>[];
      }
      rethrow;
    }

    // If another search happened after this one, throw away these options.
    if (_currentQuery != query) {
      return null;
    }
    _currentQuery = null;

    return options;
  }

  @override
  void initState() {
    super.initState();
    _debouncedSearch = _debounce<Iterable<String>?, String>(_search);
  }

  @override
  Widget build(BuildContext context) {
    return Container(
        color: Colors.red,
        height: 100,
        child: SingleChildScrollView(
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: <Widget>[
                Text(
                  _networkEnabled
                      ? 'Network is on, toggle to induce network errors.'
                      : 'Network is off, toggle to allow requests to go through.',
                ),
                Switch(
                  value: _networkEnabled,
                  onChanged: (bool? value) {
                    setState(() {
                      _networkEnabled = !_networkEnabled;
                    });
                  },
                ),
                const SizedBox(
                  height: 32.0,
                ),
                Container(
                    color: Colors.amber,
                    padding: EdgeInsets.all(5),
                    child: Autocomplete<String>(
                      fieldViewBuilder: (BuildContext context,
                          TextEditingController controller,
                          FocusNode focusNode,
                          VoidCallback onFieldSubmitted) {
                        return TextFormField(
                          decoration: InputDecoration(
                            errorText: _networkError
                                ? 'Network error, please try again.'
                                : null,
                          ),
                          controller: controller,
                          focusNode: focusNode,
                          onFieldSubmitted: (String value) {
                            onFieldSubmitted();
                          },
                        );
                      },
                      optionsBuilder: (TextEditingValue textEditingValue) async {
                        setState(() {
                          _networkError = false;
                        });
                        final Iterable<String>? options =
                        await _debouncedSearch(textEditingValue.text);
                        if (options == null) {
                          return _lastOptions;
                        }
                        _lastOptions = options;
                        return options;
                      },
                      onSelected: (String selection) {
                        debugPrint('You just selected $selection');
                      },
                    )),
              ],
            )));
  }
}

// Mimics a remote API.
class _FakeAPI {
  static const List<String> _kOptions = <String>[
    'aardvark',
    'bobcat',
    'chameleon',
  ];

  // Searches the options, but injects a fake "network" delay.
  static Future<Iterable<String>> search(
      String query, bool networkEnabled) async {
    await Future<void>.delayed(fakeAPIDuration); // Fake 1 second delay.
    if (!networkEnabled) {
      throw const _NetworkException();
    }
    if (query == '') {
      return const Iterable<String>.empty();
    }
    return _kOptions.where((String option) {
      return option.contains(query.toLowerCase());
    });
  }
}

typedef _Debounceable<S, T> = Future<S?> Function(T parameter);

/// Returns a new function that is a debounced version of the given function.
///
/// This means that the original function will be called only after no calls
/// have been made for the given Duration.
_Debounceable<S, T> _debounce<S, T>(_Debounceable<S?, T> function) {
  _DebounceTimer? debounceTimer;

  return (T parameter) async {
    if (debounceTimer != null && !debounceTimer!.isCompleted) {
      debounceTimer!.cancel();
    }
    debounceTimer = _DebounceTimer();
    try {
      await debounceTimer!.future;
    } catch (error) {
      if (error is _CancelException) {
        return null;
      }
      rethrow;
    }
    return function(parameter);
  };
}

// A wrapper around Timer used for debouncing.
class _DebounceTimer {
  _DebounceTimer() {
    _timer = Timer(debounceDuration, _onComplete);
  }

  late final Timer _timer;
  final Completer<void> _completer = Completer<void>();

  void _onComplete() {
    _completer.complete();
  }

  Future<void> get future => _completer.future;

  bool get isCompleted => _completer.isCompleted;

  void cancel() {
    _timer.cancel();
    _completer.completeError(const _CancelException());
  }
}

// An exception indicating that the timer was canceled.
class _CancelException implements Exception {
  const _CancelException();
}

// An exception indicating that a network request has failed.
class _NetworkException implements Exception {
  const _NetworkException();
}

```

</details>

**Comment by huycozy** [comment]
This is most likely related or similar to https://github.com/flutter/flutter/issues/150048

**Comment by nick9822** [comment]
@huycozy Yes it's similar may be a single fix work for both the cases, but the mentioned issue has the field in view which doesn't necessarily cause bad UI. But in my case the UI looks bad as the field goes out of view and still options in the Overlay follow.

**Comment by victorsanni** [comment]
Reopening because https://github.com/flutter/flutter/pull/154675 was reverted in https://github.com/flutter/flutter/pull/161277.

**Comment by Renzo-Olivares** [comment]
Closing this issue as a duplicate of https://github.com/flutter/flutter/issues/150048. Please feel free to re-open if this is not the case.
